kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,342 @@
+//! For now we just use simple functions to deserialize table properties from strings. This allows
+//! us to relatively simply implement the functionality described in the protocol and expose
+//! 'simple' types to the user in the [`TableProperties`] struct. E.g. we can expose a `bool`
+//! directly instead of a `BoolConfig` type that we implement `Deserialize` for.
+use std::time::Duration;
+
+use super::*;
+use crate::expressions::ColumnName;
+use crate::table_features::ColumnMappingMode;
+use crate::utils::require;
+
+const SECONDS_PER_MINUTE: u64 = 60;
+const SECONDS_PER_HOUR: u64 = 60 * SECONDS_PER_MINUTE;
+const SECONDS_PER_DAY: u64 = 24 * SECONDS_PER_HOUR;
+const SECONDS_PER_WEEK: u64 = 7 * SECONDS_PER_DAY;
+
+impl<K, V, I> From<I> for TableProperties
+where
+    I: IntoIterator<Item = (K, V)>,
+    K: AsRef<str>,
+    V: AsRef<str>,
+{
+    fn from(unparsed: I) -> Self {
+        let mut props = TableProperties::default();
+        for (k, v) in unparsed {
+            let parsed =
+                match k.as_ref() {
```

Comment:
If we pulled out the match as a separate function, we could leverage `?`:

<details>

```rust
fn try_parse(props: &mut TableProperties, k: &str, v: &str) -> Option<()> {
    match k {
        "foo" => props.foo = Some(parse_bool(v)?),
        "bar" => props.bar = Some(parse_positive_int(v)?),
          ...
        "zoom" => props.zoom = Some(IsolationLevel::try_from(v).ok()?),
        _ => return None,
    }
    Some(())
}
```
and then:
```rust
for (k, v) in unparsed {
    let (k, v) = (k.as_ref(), v.as_ref());
    if try_parse(&mut props, k, v).is_none() {
        props.unknown_properties.insert(k.to_string(), v.to_string();
    }
}
```
</details>

If we took ownership of the actual string-string map, the whole loop simplifies down to just:
```rust
unparsed.retain(|(k, v)| try_parse(&mut props, k, v).is_none());
props.unknown_properties = unparsed;
```

Reply:
Actually, we can do the ~same trick even with an iterator, it's just harder to read:

<details>

```rust
impl<K, V, I> From<I> for TableProperties
where
    I: IntoIterator<Item = (K, V)>,
    K: AsRef<str> + Into<String>,
    V: AsRef<str> + Into<String>,
{
    fn from(unparsed: I) -> Self {
        let mut props = TableProperties::default();
        let mut unparsed = unparsed
            .into_iter()
            .filter(|(k, v)| {
                // Only keep elements that fail to parse
                try_parse(&mut props, k.as_ref(), v.as_ref()).is_none()
            });
        props.unknown_properties = unparsed
            .map(|(k, v)| (k.into(), v.into()))
            .collect();
        props
    }
}
```
</details>

Reply:
oh nice let me try this :)

Reply:
done - looks good!